### PR TITLE
master_MDL-71608-56 importquestions

### PR DESCRIPTION
**Pull request by marcghaly**
PR for issue https://github.com/catalyst/moodle-MDL-70329/issues/56:

- question/bank/importquestions/classes/form/import_form.php
- question/bank/importquestions/classes/navigation.php
- question/bank/importquestions/classes/privacy/provider.php
- question/bank/importquestions/import.php
- question/bank/importquestions/lang/en/qbank_importquestions.php
- question/bank/importquestions/version.php
- lib/classes/plugin_manager.php

Code checker run : ok
core_question_testsuite: ok

### CODE REVIEW

File: question/bank/importquestions/classes/navigation.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * @package    qbank_importquestions
+ * @copyright  2021 Catalyst IT Australia Pty Ltd
+ * @author     Safat Shahin <safatshahin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class navigation extends \core_question\local\bank\navigation_plugin_base {
+
+    /**
+     * @inheritDoc
```
Comment (guillogo) [inline question/bank/importquestions/classes/navigation.php:41]: Can you add the documentation here and in all functions. Thank you.

File: question/bank/importquestions/import.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+
+require_once(__DIR__ . '/../../../config.php');
+require_once($CFG->dirroot . '/question/editlib.php');
+require_once($CFG->dirroot . '/question/bank/importquestions/classes/form/import_form.php');
+require_once($CFG->dirroot . '/question/format.php');
+require_once($CFG->dirroot . '/question/renderer.php');
+
+use qbank_importquestions\form\question_import_form;
+
```
Comment (guillogo) [inline question/bank/importquestions/import.php:33]: at some point probably here we need the call the new function that validates if a plugin is enable or not: helper::check_qbank_status(pluginname)

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @marcghaly Review done.

Can you rebase as branch master_MDL-71516 changed please. Also, that is need it as we need to call the new check_qbank_status function.

Thank you.

File: question/bank/importquestions/import.php (hunk 2 above)
Comment (safatshahin) [reply question/bank/importquestions/import.php:33]: i think it needs to be done now after rebasing with the latest 71516

File: question/bank/importquestions/classes/navigation.php (hunk 1 above)
Comment (marcghaly) [reply question/bank/importquestions/classes/navigation.php:41]: Documentation added to all methods, return type added too.

File: question/bank/importquestions/import.php (hunk 2 above)
Comment (marcghaly) [reply question/bank/importquestions/import.php:34]: Rebased and added the call

**Comment by marcghaly** [comment]
Hi @guillogo I did the change required.

File: question/bank/importquestions/classes/form/import_form.php [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+/**
+ * Form to import questions into the question bank.
+ *
+ * @copyright  2007 Jamie Pratt me@jamiep.org
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class question_import_form extends moodleform {
+
+    protected function definition() {
```
Comment (guillogo) [inline question/bank/importquestions/classes/form/import_form.php:43]: @marcghaly one last thing, that I detected in export questions, the doc here is missing. I added something like this there:

_Build the form definition._ 

_This adds all the form fields that the export questions feature needs._

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @marcghaly I've added one last little comment and after that, is ready to be merged. Thank you.

File: question/bank/importquestions/classes/form/import_form.php (hunk 3 above)
Comment (marcghaly) [reply question/bank/importquestions/classes/form/import_form.php:47]: Documentation added @guillogo

**Review by guillogo — APPROVED** [review]
Approved

**Comment by guillogo** [comment]
Hi @marcghaly this PR has been merged. Thank you.

**Pull request merged**